postgresql/src/postgresql/convert_sql.cpp

```diff
@@ -915,288 +932,412 @@ auto zpt::pgsql::get_query(zpt::json _in, std::string& _queryr) -> void {
+									_queryr += zpt::pgsql::escape_name(_key) + _op +
 										   zpt::pgsql::escape(
 										       zpt::json::string(_expression));
 								} else {
-									_queryr += zpt::pgsql::escape_name(_key) +
-										   _found->second + _bexpr;
+									_queryr += zpt::pgsql::escape_name(_key) + _op +
+										   _bexpr;
 								}
+
 							} else {
-								_queryr += zpt::pgsql::escape_name(_key) +
-									   _found->second + std::to_string(d);
+
+								_queryr += zpt::pgsql::escape_name(_key) + _op +
+									   std::to_string(d);
 							}
+
 						} else {
-							_queryr += zpt::pgsql::escape_name(_key) + _found->second +
-								   std::to_string(i);
+
+							_queryr +=
+							    zpt::pgsql::escape_name(_key) + _op + std::to_string(i);
 						}
+
 					} else if (_options == "j") {
+
 						istringstream iss(_expression);
+
 						zpt::json _json;
+
 						try {
+
 							iss >> _json;
+
 							if (_json->is_object()) {
 							} else if (_json->is_array()) {
+
 								std::string _v;
+
 								for (auto _e : _json->arr()) {
 									if (_v.length() != 0) {
 										_v += ", ";
 									}
 									_v += zpt::pgsql::escape(_e);
 								}
-								_queryr += zpt::pgsql::escape_name(_key) +
-									   _found->second + std::string("(") + _v +
-									   std::string(")");
+
+								_queryr += zpt::pgsql::escape_name(_key) + _op +
+									   std::string("(") + _v + std::string(")");
 							}
+
 						} catch (std::exception& _e) {
 						}
 
 					} else if (_options == "d") {
-						_queryr += zpt::pgsql::escape_name(_key) + _found->second +
+
+						_queryr += zpt::pgsql::escape_name(_key) + _op +
 							   std::string("TIMESTAMP('") +
 							   zpt::pgsql::escape(zpt::json::string(_expression)) +
 							   std::string("')");
 					}
+
 					continue;
 				}
 			}
 		}
 
-		_queryr += zpt::pgsql::escape_name(_key) + std::string("=") + zpt::pgsql::escape(_v);
+		_queryr += zpt::pgsql::escape_name(_key) + std::string(" = ") + zpt::pgsql::escape(_v);
 	}
 }
 
 auto zpt::pgsql::get_opts(zpt::json _in, std::string& _queryr) -> void {
+
 	if (!_in->is_object()) {
 		return;
 	}
 
-	if (_in["page_size"]) {
-		_queryr += std::string(" LIMIT ") + std::to_string(int(_in["page_size"]));
-	}
-	if (_in["page_start_index"]) {
-		_queryr += std::string(" OFFSET ") + std::to_string(int(_in["page_start_index"]));
-	}
 	if (_in["order_by"]) {
+
 		_queryr += std::string(" ORDER BY ");
+
 		std::istringstream lss(((std::string)_in["order_by"]).data());
 		std::string _part;
+
 		bool _first = true;
+
 		while (std::getline(lss, _part, ',')) {
+
 			if (_part.length() > 0) {
+
 				std::string _dir = "ASC";
 
 				if (_part[0] == '-') {
 					_dir = "DESC";
 				}
+
 				_part.erase(0, 1);
-				_queryr += (!_first ? ", " : "") + zpt::pgsql::escape(zpt::json::string(_part)) +
+
+				_queryr += (!_first ? ", " : "") +
+					   zpt::r_replace(zpt::pgsql::escape(zpt::json::string(_part)), "'", "\"") +
```

Comment:
Could we do this in a more general manner?
Change `zpt::pgsql::escape` to take a second argument, an `std::string`, with default value `'`. The method signature would become:
```c++
auto escape(zpt::json _in, std::string _str_delimiter = "'") -> std::string;
```
and the function body would become:
```c++
auto zpt::pgsql::escape(zpt::json _in, std::string _str_delimiter) -> std::string {
	std::string _out;

	switch (_in->type()) {
	case zpt::JSObject: {
		_out.assign(zpt::pgsql::escape(zpt::json::string(std::string(_in))));
		break;
	}
	case zpt::JSArray: {
		if (_in->arr()->size() == 2 && _in[0]->is_number() && _in[1]->is_number()) {
			_out.assign(std::string("point") +
				    zpt::r_replace(zpt::r_replace(std::string(_in), "[", "("), "]", ")"));
		} else {
			_out.assign(zpt::pgsql::escape(zpt::json::string(std::string(_in))));
		}
		break;
	}
	case zpt::JSString: {
		_out.assign(std::string(_in));
		zpt::replace(_out, _str_delimiter, str_delimiter + str_delimiter);
		_out.insert(0, str_delimiter);
		_out.push_back(str_delimiter{0});
		break;
	}
	case zpt::JSBoolean: {
		_out.assign(std::string(_in));
		break;
	}
	case zpt::JSDouble: {
		_out.assign(std::string(_in));
		break;
	}
	case zpt::JSInteger: {
		_out.assign(std::string(_in));
		break;
	}
	case zpt::JSNil: {
		_out.assign("NULL");
		break;
	}
	case zpt::JSDate: {
		_out.assign(std::string("TIMESTAMP ") + _str_delimiter + std::string(_in) + _str_delimiter);
		break;
	}
	case zpt::JSLambda: {
		_out.assign(std::string(_in));
		break;
	}
	default: {}
	}

	return _out;
}
```